internal/js_scanner/js_scanner.go

```diff
@@ -167,6 +228,42 @@ func readLineComment() {
 	}
 }
 
+func readTemplateString() {
```

Comment:
Are these pulled from somewhere? Or did you write them?

Reply:
Reason I ask is because I’d like to either link to documentation where these originated, or I think we should add more comments here (such as examples of what each function is / isn’t meant to handle)